### PR TITLE
Change logger to use formatter

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Add two formatting classes for stream and logfile

Add filter to apply additional labels based upon logging level

Signed-off-by: brentholtsclaw <brent.holtsclaw@intel.com>

### CODE REVIEW

**Review by npmitche — CHANGES_REQUESTED** [review]

**Comment by npmitche** [comment]
I'm seeing what @jun2zhou posted on PR #1608:
 https://github.com/chipsec/chipsec/pull/1608#issuecomment-1387166955 

with this PR. Is that expected?

File: chipsec/logger.py [hunk 1]
```diff
@@ -345,22 +382,22 @@ def log_information(self, text):    # Use log("text")
...
     # -----------------------------
 
-    def start_test(self, test_name:str) -> None:
+    def start_test(self, test_name: str) -> None:
         """Logs the start point of a Test"""
         text = '[x][ =======================================================================\n'
         text = text + '[x][ Module: ' + test_name + '\n'
         text = text + '[x][ ======================================================================='
-        self.chipsecLogger.info(text, 'BLUE')
+        self.chipsecLogger.info(text, level.INFO, 'BLUE')
```
Comment (npmitche) [inline chipsec/logger.py:390]:
```suggestion
        self.log(text, level.INFO, 'BLUE')
```

File: chipsec/logger.py [hunk 2]
```diff
@@ -155,53 +210,31 @@ def __init__(self, profile='debug.conf'):
...
         self.chipsecLogger.addHandler(fileH)
-        logging.addLevelName(11, "VERBOSE")
-        logging.addLevelName(12, "HAL")
-        logging.addLevelName(13, "HELPER")
-        logging.setLogRecordFactory(chipsecRecordFactory)  # applies colorization to output
+        self.chipsecLogger.addFilter(chipsecFilter(LOGGER_NAME))
+        logging.addLevelName(level.VERBOSE.value, level.VERBOSE.name)
+        logging.addLevelName(level.HAL.value, level.HAL.name)
+        logging.addLevelName(level.HELPER.value, level.HELPER.name)
+        logFormatter = chipsecLogFormatter('%(additional)s %(message)s')
```
Comment (npmitche) [inline chipsec/logger.py:217]:
```suggestion
        logFormatter = chipsecLogFormatter('%(additional)s%(message)s')
```
Additional space adds a space to the beginning if there's no prefix.

File: chipsec/logger.py [hunk 3]
```diff
@@ -155,53 +210,31 @@ def __init__(self, profile='debug.conf'):
...
+        streamFormatter = chipsecStreamFormatter('%(additional)s %(message)s')
```
Comment (npmitche) [inline chipsec/logger.py:218]: Same as above.

**Review by npmitche — CHANGES_REQUESTED** [review]

File: chipsec/logger.py [hunk 4]
```diff
@@ -51,14 +51,77 @@
...
+            record.additional = '[*] [HAL] '
+        elif record.levelno == level.HELPER.value:
+            record.additional = '[*] [HELPER] '
+        else:
+            record.additional = ''
+        return True
+
+
+class chipsecLogFormatter(logging.Formatter):
+    def __init__(self, fmt: Optional[T] = ..., datefmt: Optional[T] = ..., style='%') -> None:
```
Comment (npmitche) [inline chipsec/logger.py:99]:
```suggestion
    def __init__(self, fmt: Optional[str] = ..., datefmt: Optional[str] = ..., style='%') -> None:
```
Sorry, T was just a placeholder. Please keep the correct type. Everything else is looking good!

File: chipsec/logger.py [hunk 5]
```diff
@@ -51,14 +51,77 @@
...
+
+    def format(self, record):
+        if record.args:
+            record.args = tuple()
+        formatter = logging.Formatter(self.infmt)
+        return formatter.format(record)
+
+
+class chipsecStreamFormatter(logging.Formatter):
+    def __init__(self, fmt: Optional[T] = ..., datefmt: Optional[T] = ..., style='%') -> None:
```
Comment (npmitche) [inline chipsec/logger.py:111]:
```suggestion
    def __init__(self, fmt: Optional[str] = ..., datefmt: Optional[str] = ..., style='%') -> None:
```

**Review by npmitche — CHANGES_REQUESTED** [review]

**Review by npmitche — APPROVED** [review]

**Review by dscott90 — APPROVED** [review]

**Pull request merged**